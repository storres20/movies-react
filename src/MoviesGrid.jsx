import { MovieCard } from "./MovieCard";
import movies from "./movies.json";
import styles from "./MoviesGrid.module.css";

export function MoviesGrid() {
  console.log(movies);
  return (
    <ul className={styles.moviesGrid}>
      {movies.map((movie) => {
        return <MovieCard key={movie.id} movie={movie} />
      })}
    </ul>
    );
}