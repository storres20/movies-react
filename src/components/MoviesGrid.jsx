import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import MovieCard from './MovieCard';
import styles from './MoviesGrid.module.css';
import ScrollButton from './ScrollButton';
// Redux
import { getMovie } from '../redux/movies/movies';
import SearchBar from './SearchBar';
// ***

export default function MoviesGrid() {
  /* Setup Redux dispatch */
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMovie());
  }, [dispatch]);

  /* Load Redux State */
  const movies = useSelector((state) => state.movie);

  // *********************************

  return (
    <div>
      {/* Filter Search Bar */}
      <SearchBar />

      <ul className={styles.moviesGrid}>
        {movies.map((movie) => <MovieCard key={movie.id} movie={movie} />)}
      </ul>

      {/* Scroll button for going back to TOP page */}
      <ScrollButton />
    </div>
  );
}
