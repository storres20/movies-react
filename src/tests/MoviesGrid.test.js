import React from 'react';
import renderer from 'react-test-renderer';
import MoviesGrid from '../components/MoviesGrid';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MoviesGrid />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
