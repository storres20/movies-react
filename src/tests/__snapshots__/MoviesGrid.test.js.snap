// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders correctly 1`] = `
<div>
  <div>
    <div
      className="divFlex"
    >
      <input
        className="inputFlex"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
      />
    </div>
  </div>
  <ul
    className="moviesGrid"
  />
  <div
    className="button"
  />
</div>
`;
